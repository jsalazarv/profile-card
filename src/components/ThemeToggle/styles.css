.toggle:checked {
    @apply appearance-none;
    @apply outline-none;
    @apply cursor-pointer;
    @apply w-8;
    @apply h-8;
    @apply rounded-full;
    @apply text-white;
    @apply scale-75;
    transition: all 500ms;
    box-shadow: inset calc(var(--size) * 0.33) calc(var(--size) * -0.25) 0;
}

.toggle {
    @apply appearance-none;
    @apply outline-none;
    @apply cursor-pointer;
    @apply w-8;
    @apply h-8;
    @apply rounded-full;
    @apply text-amber-500;
    @apply scale-75;
    @apply bg-black;
    @apply border;
    transition: all 500ms;
    box-shadow: inset 0 0 0 var(--size), calc(var(--offset-orthogonal) * -1) 0 0 var(--ray-size), var(--offset-orthogonal) 0 0 var(--ray-size), 0 calc(var(--offset-orthogonal) * -1) 0 var(--ray-size), 0 var(--offset-orthogonal) 0 var(--ray-size), calc(var(--offset-diagonal) * -1) calc(var(--offset-diagonal) * -1) 0 var(--ray-size), var(--offset-diagonal) var(--offset-diagonal) 0 var(--ray-size), calc(var(--offset-diagonal) * -1) var(--offset-diagonal) 0 var(--ray-size), var(--offset-diagonal) calc(var(--offset-diagonal) * -1) 0 var(--ray-size);
}