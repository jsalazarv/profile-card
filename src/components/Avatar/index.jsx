import './styles.css';
import { useMemo } from 'react';

export const Avatar = ({
  className = '',
  children,
  size = 'small',
  type = 'circle',
  color = 'bg-primary',
}) => {
  const sizes = {
    small: 'w-6 h-6',
    medium: 'w-16 h-16',
    large: 'w-32 h-32',
  };

  const types = {
    circle: 'rounded-full',
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };

  const avatarSize = useMemo(() => sizes[size], [size]);
  const avatarType = useMemo(() => types[type], [type]);

  return (
    <div
      className={`overflow-auto ${avatarSize} ${avatarType} ${color} ${className}`}>
      {children}
    </div>
  );
};
