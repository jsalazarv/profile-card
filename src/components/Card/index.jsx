import './styles.css';
import { useMemo } from 'react';

export const Card = ({
  className = '',
  children,
  type = 'tile',
  outlined = '',
}) => {
  const types = {
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };
  const outline = 'border-solid border';
  const cardType = useMemo(() => types[type], [type]);

  return (
    <div className={`card ${cardType} ${outlined && outline} ${className}`}>
      {children}
    </div>
  );
};
