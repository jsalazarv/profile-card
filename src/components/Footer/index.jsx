import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import './styles.css';

export const Footer = ({ phrase = 'Designed with' }) => {
  return (
    <>
      <footer className="footer">
        <p>
          {phrase}
          <span>
            <FontAwesomeIcon icon={faHeart} />
          </span>
        </p>
      </footer>
    </>
  );
};
