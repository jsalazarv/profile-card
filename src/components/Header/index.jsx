import './styles.css';
import { ThemeToggle } from '../ThemeToggle/index.jsx';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

export const Header = ({}) => {
  return (
    <header className="header">
      <nav className="header__navbar" aria-label="Top">
        <div className="navbar__content">
          <div className="logo__container">
            <a href="https://jsalazarv.dev/" target="_blank">
              <h2>[Js]</h2>
            </a>
          </div>
          <div className="items__container">
            <a href="https://github.com/jsalazarv/profile-card" target="_blank">
              <FontAwesomeIcon
                className="text-2xl border-black dark:text-amber-500 dark:text-3xl transition-all duration-1000"
                icon={faGithub}
              />
            </a>
            <ThemeToggle />
          </div>
        </div>
      </nav>
    </header>
  );
};
