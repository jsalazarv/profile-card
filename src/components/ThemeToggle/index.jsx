import './styles.css';
import useDarkMode from '../../hooks/useDarkMode.js';

export const ThemeToggle = ({ className = '' }) => {
  const [colorTheme, setTheme] = useDarkMode();

  const toggleTheme = () => {
    setTheme(colorTheme);
  };

  return (
    <input
      className={`toggle ${className}`}
      type="checkbox"
      onClick={toggleTheme}
      aria-label="toggleTheme"
    />
  );
};
