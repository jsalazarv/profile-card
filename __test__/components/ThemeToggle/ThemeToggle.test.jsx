import { fireEvent, screen, render } from '@testing-library/react';
import { ThemeToggle } from '../../../src/components/ThemeToggle';

const root = window.document.documentElement;

describe('[ThemeToggle Component]', () => {
  describe('[Snapshots]', () => {
    test('It should contain the class [toggle]', () => {
      const { container } = render(<ThemeToggle />);

      expect(container.firstChild).toHaveClass('toggle');
      expect(container).toMatchSnapshot();
    });

    test('It should have dark class the first time it loads', () => {
      expect(root).toHaveClass('dark');
    });

    test('It should have light class the first time it clicked', () => {
      render(<ThemeToggle />);

      fireEvent.click(screen.getByRole('checkbox', { name: 'toggleTheme' }));
      expect(root).toHaveClass('light');
    });

    test('It should have light class the second time it clicked', () => {
      render(<ThemeToggle />);

      fireEvent.click(screen.getByRole('checkbox', { name: 'toggleTheme' }));
      fireEvent.click(screen.getByRole('checkbox', { name: 'toggleTheme' }));
      expect(root).toHaveClass('dark');
    });
  });
});
