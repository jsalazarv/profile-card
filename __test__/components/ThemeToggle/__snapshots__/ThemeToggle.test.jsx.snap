// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ThemeToggle Component] [Snapshots] It should contain the class [toggle] 1`] = `
<div>
  <input
    aria-label="toggleTheme"
    class="toggle "
    type="checkbox"
  />
</div>
`;
